core/random.rbs

```diff
@@ -104,14 +31,16 @@ class Random < Object
   #
   def self.new_seed: () -> Integer
 
-  # Alias of Random::DEFAULT.rand.
-  #
   def self.rand: () -> Float
-               | (Integer | ::Range[Integer] max) -> Integer
-               | (Float | ::Range[Float] max) -> Float
+               | (Integer max) -> Integer
+               | (Float max) -> Float
+               | (::Range[Integer] range) -> Integer?
+               | (::Range[Float] range) -> Float?
+
```

Comment:
The original had a definition for `Numeric`, but I am not sure if it is needed.

Reply:
In what cases does it return `nil` in `Random.rand`?
The last check seems to have eliminated the possibility of returning nil or false.

https://github.com/ruby/ruby/blob/342e7a094a70d6f90b96262c88177dae32976c85/random.c#L1388-L1389

There is no such description in ruby/spec either.

https://github.com/ruby/spec/blob/a8afb240adcb413994d4de4d33f3f725105811da/core/random/rand_spec.rb

Reply:
Thank you for your comment and for the helpful information.

As for the part you asked about, I referred to the following document.

https://docs.ruby-lang.org/ja/3.0/method/Random/i/rand.html

I have read the above and confirmed that there is a difference between RDoc and the description.
However, I was not able to follow the C implementation.

So, I wrote the RBS according to the above document for the time being.